### ISSUE TITLE
Wifi not connecting

### ISSUE THREAD

**Issue by Brakelmann** [jomjol/AI-on-the-edge-device#477]
Hi,

my ESP32 CAM device is not connecting to my WIFI.
I tried different AccessPoints to connect to without success to connect to the WIFI.

_17:02:36.921 -> open file /sdcard/wlan.ini in mode r
17:02:36.954 -> file loaded
17:02:36.954 -> 
17:02:36.954 -> WLan: „VWLANGast“, „hallogast“
17:02:36.954 -> Hostename: watermeter
17:02:36.954 -> I (2899) wifi:wifi driver task: 3ffd3590, prio:23, stack:6656, core=0
17:02:36.954 -> [0;32mI (2899) system_api: Base MAC address is not set, read default base MAC address from BLK0 of EFUSE[0m
17:02:36.991 -> [0;32mI (2899) system_api: Base MAC address is not set, read default base MAC address from BLK0 of EFUSE[0m
17:02:37.161 -> I (3089) wifi:wifi firmware version: 3ea4c76
17:02:37.161 -> I (3099) wifi:config NVS flash: enabled
17:02:37.161 -> I (3099) wifi:config nano formating: disabled
17:02:37.161 -> I (3099) wifi:Init dynamic tx buffer num: 32
17:02:37.161 -> I (3099) wifi:Init data frame dynamic rx buffer num: 32
17:02:37.161 -> I (3099) wifi:Init management frame dynamic rx buffer num: 32
17:02:37.195 -> I (3109) wifi:Init management short buffer num: 32
17:02:37.195 -> I (3109) wifi:Init static tx buffer num: 16
17:02:37.195 -> I (3119) wifi:Init static rx buffer size: 1600
17:02:37.195 -> I (3119) wifi:Init static rx buffer num: 10
17:02:37.195 -> I (3129) wifi:Init dynamic rx buffer num: 32
17:02:37.264 -> [0;33mW (3199) phy_init: failed to load RF calibration data (0x1102), falling back to full calibration[0m
17:02:37.401 -> [0;32mI (3349) phy: phy_version: 4180, cb3948e, Sep 12 2019, 16:39:13, 0, 2[0m
17:02:37.676 -> I (3599) wifi:mode : sta (7c:9e:bd:45:e5:f4)
17:02:37.676 -> [0;32mI (3599) BLINK: Blinken - start[0m
17:02:37.676 -> [0;32mI (3599) wifi station: wifi_init_sta finished.[0m
17:02:37.846 -> [0;32mI (3799) BLINK: Blinken - done[0m
17:02:39.702 -> [0;32mI (5649) wifi station: retry to connect to the AP[0m
17:02:39.735 -> [0;32mI (5659) wifi station: connect to the AP fail[0m
17:02:41.774 -> [0;32mI (7709) wifi station: retry to connect to the AP[0m
17:02:41.774 -> [0;32mI (7709) wifi station: connect to the AP fail[0m
17:02:43.828 -> [0;32mI (9759) wifi station: retry to connect to the AP[0m
17:02:43.828 -> [0;32mI (9759) wifi station: connect to the AP fail[0m
17:02:45.869 -> [0;32mI (11809) wifi station: retry to connect to the AP[0m_

Using the ESP32 CAM WebServer sample connects to the same WIFI without any problems.
_17:07:58.807 -> .......
17:08:01.789 -> WiFi connected
17:08:01.789 -> Starting web server on port: '80'
17:08:01.789 -> Starting stream server on port: '81'
17:08:01.789 -> Camera Ready! Use 'http://192.168.179.22' to connect_

Any idea or help would be much appreciated.
Thanks, Stefan.

**Comment by jomjol** [comment]
Which protocol version do you use in your WLAN?

**Comment by Brakelmann** [comment]
The fritzBox is configured to use automatic settings for WLAN.
But I guess its been 802.11 g&n if it set to auto config.
![Bildschirmfoto 2021-12-24 um 10 23 52](https://user-images.githubusercontent.com/32041481/147340096-4941c9c9-70dd-47ea-9e98-16d41a227a12.png)

**Comment by jomjol** [comment]
Sorry, no idea abou it. I'm using the standard ESP32 WLan configuration and your reporting for the first time this kind of problem.
Maybe you have a special character in the password, that could cause troube?

**Comment by jomjol** [comment]
Did you solve the problem? Meanwhile I will close this. 

Please reopen on demand.

**Comment by Brakelmann** [comment]
No I don't, ordered new ESP32 Cam devices and will report it will work with the other devices.

**Comment by Brakelmann** [comment]
I got two new ESP32 CAM devices. Both show the same behavior.
The boot process stucks at the wifi connection.
I tried it with three different ESP32 Cam devices with 3 differen Access Points (Fritz.box 7490, TPLInk and Wifi Tethering on mobile phone).
I tried with different SSID Names and with different password, no luck.

Is there any chance to get more debug information on the wifi connection procedure?

I only get the folloing messages without error codes...
␛[0;32mI (347877) wifi station: connect to the AP fail␛[0m
␛[0;32mI (349927) wifi station: retry to connect to the AP␛[0m
␛[0;32mI (349927) wifi station: connect to the AP fail␛[0m
␛[0;32mI (351977) wifi station: retry to connect to the AP␛[0m

**Comment by Brakelmann** [comment]
No need to reopen.
I got it solved. Assuming the configuriation values need to have "" I configured the values like...
ssid = "Keller"
password = "Keller1234"
hostname = watermeter

The values must not have "" so I got it working like this...
ssid = Keller
password = Keller1234
hostname = watermeter

Thanks for your effort and this fantastic projekt.
Now I need to Print my cam holder and get it all installed.

Best regards, Stefan.
